packages/react-components/react-provider/src/components/FluentProvider/useFluentProviderThemeStyleTag.ts

```diff
@@ -3,25 +3,47 @@ import * as React from 'react';
 import type { FluentProviderState } from './FluentProvider.types';
 import { fluentProviderClassNames } from './useFluentProviderStyles';
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const useInsertionEffect = (React as any)['useInsertion' + 'Effect'];
```

Comment:
I think we should add a comment here that explains [why we write this as `['useInsertion' + 'Effect']`](https://github.com/emotion-js/emotion/pull/2651).

Reply:
to be honest, i'm not sure if it is necessary as this repo has different rules than griffel

Reply:
In that case it's worth experimenting with a CRA app to see if we need to do this. In the linked Emotion issue the problem didn't pop up in Emotion but in a consumer of Emotion using React 17.

Reply:
It will cause problems on consumer side, please check https://github.com/microsoft/fluentui/issues/23625#issuecomment-1193376468